### PR TITLE
[BugFix] Get the inconsistent backup snapshot cause by the concurrent between data loading and backup (#33426)

### LINKED ISSUE #33426: The restored table can not be loaded new data any more
In some case, the restored table can not be loaded new data any more. If the snapshot is backuped when the table is being loaded.

### PR DESCRIPTION
**Pull request by srlch**
Problem:
If the data loading and backup executing concurrently, the backuped snapshot for table metadata will get inconsistent partition verison info that version + 2 = nextversion, but the correct version info should be version + 1 = nextversion. If we restore such snapshot, we will get the table and it is can not be loading data any more for primary key table.

Solution:
check the version info when backuping the table meta info.

Fixes #33426

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/backup/BackupJob.java [hunk 1]
```diff
@@ -513,6 +513,21 @@ private void prepareAndSendSnapshotTask() {
...
 
+        // check parittion version for backupMeta
+        Map<String, Table> tablesMapInBackupMeta = backupMeta.getTables();
+        for (Map.Entry<String, Table> entry : tablesMapInBackupMeta.entrySet()) {
+            OlapTable olapTable = (OlapTable) entry.getValue();
+            for (Partition part : olapTable.getPartitions()) {
+                if (part.getNextVersion() != part.getVisibleVersion() + 1) {
+                    status = new Status(ErrCode.COMMON_ERROR, "backuped partition in table: " + olapTable.getName() +
+                                        "has inconsistent state for parition " + part.getName() +
+                                        "please stop ingesting data into table: " + olapTable.getName() +
```
Comment (trueeyu) [inline fe/fe-core/src/main/java/com/starrocks/backup/BackupJob.java:524]: "stop ingesting data into table" why ?

File: fe/fe-core/src/main/java/com/starrocks/backup/BackupJob.java [hunk 2]
```diff
@@ -513,6 +513,21 @@ private void prepareAndSendSnapshotTask() {
             db.readUnlock();
         }
 
+        // check parittion version for backupMeta
+        Map<String, Table> tablesMapInBackupMeta = backupMeta.getTables();
+        for (Map.Entry<String, Table> entry : tablesMapInBackupMeta.entrySet()) {
+            OlapTable olapTable = (OlapTable) entry.getValue();
+            for (Partition part : olapTable.getPartitions()) {
+                if (part.getNextVersion() != part.getVisibleVersion() + 1) {
```
Comment (trueeyu) [inline fe/fe-core/src/main/java/com/starrocks/backup/BackupJob.java:521]: Backup just gets a snapshot, why do we need to use retry to achieve it?

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 7 / 11 (63.64%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|7|11|63.64%|[522, 523, 524, 526]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Pull request closed without merging**